tutorials/intermediate/arranging_axes.py

```diff
@@ -0,0 +1,314 @@
+
+############################################################################
+# High-level methods for making grids
+# ===================================
+#
+# Basic 2x2 grid
+# --------------
+#
+# We can create a basic 2-by-2 grid of axes using
+# :func:`~matplotlib.pyplot.subplots`.  It returns a
+# :class:`~matplotlib.figure.Figure` instance and an array of
+# :class:`~matplotlib.axes.Axes` objects.  The axes objects can
+# be used to access methods to place artists on the axes; here we
+# use `~.Axes.annotate`, but other examples could be `~.Axes.plot`,
+# `~.Axes.pcolormesh`, etc.
+
+fig, axs = plt.subplots(ncols=2, nrows=2, figsize=(4.5, 3.5),
+                        constrained_layout=True)
+for row in range(2):
+    for col in range(2):
+        axs[row, col].annotate(f'axs[{row}, {col}]', (0.1, 0.5),
+                               xycoords='axes fraction', va='center')
+fig.suptitle('subplots')
+
+##############################################################################
+# The same effect can be achieved with `~.pyplot.subplot_mosaic`,
+# but the return type is a dictionary instead of an array, where the user
+# can give the keys useful meanings.  Here we provide two lists, each list
+# representing a row, and each element in the list a key representing the
+# column.  Note that keys can be any dictionary key, but we typically use
+# strings:
+
+fig, axs = plt.subplot_mosaic([['upleft', 'upright'], ['loleft', 'loright']],
+                              figsize=(4.5, 3.5), constrained_layout=True)
+for k in axs.keys():
+    axs[k].annotate(f'axs["{k}"]', (0.1, 0.5),
+                    xycoords='axes fraction', va='center')
+fig.suptitle('subplot_mosaic')
+
+############################################################################
+# Axes spanning rows or columns in a grid
+# ---------------------------------------
+#
+# Sometimes we want axes to span rows or columns of the grid.
+# There are actually multiple ways to accomplish this, but the most
+# convenient is probably to use `~.pyplot.subplot_mosaic` by repeating one
+# of the keys:
+
+fig, axs = plt.subplot_mosaic([['upleft', 'right'], ['loleft', 'right']],
+                              figsize=(4.5, 3.5), constrained_layout=True)
+for k in axs.keys():
+    axs[k].annotate(f'axs["{k}"]', (0.1, 0.5),
+                    xycoords='axes fraction', va='center')
+fig.suptitle('subplot_mosaic')
+
+############################################################################
+# See below for the description of how to do the same thing using
+# `~matplotlib.gridspec.GridSpec` or ~matplotlib.pyplot.subplot2grid`.
+#
+# Variable widths or heights in a grid
+# ------------------------------------
+#
+# Both `~.pyplot.subplots` and `~.pyplot.subplot_mosaic` allow the rows
+# in the grid to be different heights, and the columns to be different
+# widths using the *gridspec_kw* keyword argument.
+# Any parameter accepted by :class:`~matplotlib.gridspec.GridSpec` can
```

Comment:
Technically, not `nrows`, `ncols`, and `figure`, but that might be too pedantic. The broader question: would it be worthwhile to positively list the supported paramters here? On the one hand, that's instructive, on the other, that may be too much detail.

Reply:
Changed to 
```
# Spacing parameters accepted by :class:`~matplotlib.gridspec.GridSpec`
# can be passed to
```